samples: cmsis_rtos_v1: Samples to demo usage of cmsis_rtos_v1

spoorthik opened the pull request:
Add samples to demonstrate usage of cmsis_rtos_v1 APIs with dining
philosopher's problem implementation and basic message queue usage
with timers.

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

dbkinder commented on samples/portability/cmsis_rtos_v1/philosophers/README.rst:
```diff
@@ -0,0 +1,63 @@
+    Philosopher 4 [P: 2]  THINKING [  800 ms ]
+    Philosopher 5 [P: 3]   EATING  [  625 ms ]
+
+
```
Delete most of these blank lines

dbkinder commented on samples/portability/portability.rst:
```diff
@@ -0,0 +1,12 @@
+.. toctree::
+   :maxdepth: 2
+   :glob:
+   
```
Checkpatch isn't happy about trailing spaces on this line
```suggestion

```

rgundi commented on samples/portability/cmsis_rtos_v1/timer_synchronization/src/main.c:
```diff
@@ -0,0 +1,84 @@
+	send_msg_thread();
+	osTimerStart(timer_id, TIMER_TICKS);
+
+	while (counter--) {
```
If you use --counter here instead of counter--, you can check for counter == 0 in line number 79.

rgundi commented on samples/portability/portability.rst:
```diff
@@ -8,5 +8,3 @@ Portability Samples
    :glob:
 
    **/*
-
```
Not sure what this file is there for. You are only deleting empty lines here in this commit.

spoorthik committed:
samples: cmsis_rtos_v1: Demo with Dining philosopher's problem

Sample application to demonstrate usage of cmsis_rtos_v1 APIs
with dining philosopher's problem implementation.

This covers semaphores, mutex and thread APIs of CMSIS RTOS V1.

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

spoorthik committed:
samples: cmsis_rtos_v1: Demo on msgq and timers

Sample to demonstrate usage of message queue, threads
and timers with CMSIS RTOS V1 APIs.

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

spoorthik commented:
@rgundi: Thanks Raj, I have addressed your review comments. Please have a look now.

rgundi reviewed (approved):
LGTM

spoorthik commented:
Thanks @rgundi. 
@dbkinder: Can you please review on the changes made as you suggested?

dbkinder reviewed (approved):
Doc changes LGTM, thanks.

SebastianBoe commented:
Hi, can we revert this?

@spoorthik , @nashif 

It appears the sample fails intermittently:

https://github.com/zephyrproject-rtos/zephyr/pull/11930

galak commented:
Not sure if we should revert, or just not run on qemu_xtensa

galak commented:
> Not sure if we should revert, or just not run on qemu_xtensa

Also, I wondering does cmsis_rtos_v{1,2} make sense on non-ARM platforms?

galak commented:
Opened #11936
